pippy/PipelineSchedule.py

```diff
@@ -603,27 +603,22 @@ def step_microbatches(
         losses: Optional[List] = None,
     ):
         """
-        # n_loop = n_stage / n_pp
-        # run microbatches in sequences of NPp
+        Operate on the microbatches for interleaved 1f1b schedule (https://arxiv.org/pdf/2104.04473.pdf).
 
-        schedule operates at the rank level
-
-        highest rank has a warmup (F only) count of [len(stages) - 1] * seq_size
-        each hop away from highest rank adds 2 warmup stages
+        Highest rank has a warmup (F only) count of [len(stages) - 1] * group_size
+        and each hop away from highest rank adds 2 warmup stages due to:
             - one happened before highest rank's warmup started,
             - one waiting for backward result to trickle down from highest rank
-        dist_from_highest = (worldsize - 1) - rank
-
-        total_steps = warmup_steps + (num_stages * num_microbatch)
 
-        Rank 0: 0F 0F 0F 0F 2F 2F 2F 2F
-        Rank 1:    1F 1F 1F 1F 3F3B 3F 3F 3F
+        TODO: Interleaved 1F1B does not support using sorted_batch_isend_irecv()
+        because it requires recvs and sends from different peers
+        to execute in the same coalesced operation. As a result, this schedule does
+        not support models with skip connections.
```

Comment:
I like this comment. To actuate it, we can add a `has_skip_conn` filed in `pipe_info`, so that the schedule can check this field and error out here. (Rather than silently hang)